pytorch_lightning/loops/dataloader/evaluation_dataloader_loop.py

```diff
@@ -99,6 +101,23 @@ def reset(self) -> None:
         if isinstance(self._max_batches, int):
             self._max_batches = [self._max_batches] * len(self.dataloaders)
 
+    def run(self, *args: Any, **kwargs: Any) -> Optional[Any]:
+        if not (self.trainer.evaluating or self.trainer.sanity_checking):
+            rank_zero_warn(
+                f"`validation` was called but the running stage is set to {self.trainer.state.stage}."
+                " This should not happen normally. Setting it to `RunningStage.VALIDATING`", RuntimeWarning
+            )
+            self.validating = True
```

Comment:
I added this back when I did the runningstage refactor just in case, but nothing has come up

Maybe we could change it now to an `assert`

Reply:
Done !